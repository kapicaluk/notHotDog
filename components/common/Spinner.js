import ActivityIndication from 'react-native'

const Spinner = () => <ActivityIndicator size="small" />
