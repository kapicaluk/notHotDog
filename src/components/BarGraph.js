import React from 'react';
import { Text, View } from 'react-native';
import { VictoryBar, VictoryLabel } from 'victory-native';

const BarGraph = (props) => {

  const nutrients = props.nutrients ? props.nutrients.dv : null
  let nutrientNames = Object.keys(nutrients)
  let data = [];
  if (Array.isArray(nutrients)) {
    nutrients.forEach(nutrient => {
      data.push({ x: nutrient.name, y: nutrient.percentdv })
    });
  } else {
    nutrientNames.forEach(name => {
      data.push({x: name, y: nutrients[name]})
    })
  }

  return (
    <View style={{alignItems: 'center', paddingTop: 15}}>
      <Text style={{alignSelf: 'center', fontWeight: 'bold', fontSize: 14, color: 'grey'}}>Nutrient Percent Daily Values</Text>
      <View style={{paddingTop: -15}}>
        <VictoryBar
          horizontal={true}
          data={data}
          height={1000}
          labels={(data) => `${data.x} - ${data.y}%`}
          style={{data: { fill: 'tomato', width: 15 }}}
          animate={{duration: 0, onLoad: {duration: 0}}}
          labelComponent={<VictoryLabel x={50} />}
        />
      </View>
    </View>
  )
}

export default BarGraph;
