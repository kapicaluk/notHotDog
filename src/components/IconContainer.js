import React from 'react';
import { Icon } from 'react-native-elements';
import { Actions } from 'react-native-router-flux';
import { View, StyleSheet } from 'react-native';

const IconContainer = () => {
    return (
        <View style={styles.iconContainer}>
            <Icon raised name="camera" color='#00aced' reverse onPress={() => { Actions.gallery() }} />
            <Icon raised name="settings" color='#517fa4' reverse onPress={() => { console.log('going to setting page') }}/>
        </View>
    )
}

const styles = StyleSheet.create({
    iconContainer: {
        flexDirection: 'row',
        justifyContent: 'flex-end'
    }
});

export default IconContainer;