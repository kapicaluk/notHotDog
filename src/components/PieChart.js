import React, { Component } from "react";
import { Text, StyleSheet, View, ScrollView, Dimensions } from "react-native";
import { VictoryPie, VictoryLegend } from "victory-native";
import { Svg } from "react-native-svg";
import { Icon, List, ListItem, Card } from "react-native-elements";

class PieChart extends Component {
 
  render() {
    const unformatted = this.props.allFoods.data
    let chartData = [];
    chartData.push(
        { x: "carbs", y: unformatted.carbs },
        { x: "fat", y: unformatted.fat },
        { x: "protein", y: unformatted.protein }
    );
    return (
    
      <View style={styles.container}>
      <View style={styles.pie}>
        <VictoryPie
          colorScale={["tomato", "orange", "gold"]}
          data={chartData}
          innerRadius={100}
          padAngle={2}
          style={{ labels: { fontSize: 1 } }}
        />
      </View>
      <Card title="Source of Calories" containerStyle={styles.legend}>
        <View flexDirection="row">
          <View style={styles.legendItem} flexDirection="row"><Icon marginRight={2} name="label" color="tomato"/><Text style={styles.legendText}>Carbs</Text></View>
          <View style={styles.legendItem} flexDirection="row"><Icon marginRight={2} name="label" color="orange"/><Text style={styles.legendText}>Fat</Text></View>
          <View style={styles.legendItem} flexDirection="row"><Icon marginRight={2} name="label" color="gold"/><Text style={styles.legendText}>Protein</Text></View>
        </View>
      </Card>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    alignItems: "center",
    justifyContent: "center",
    paddingLeft: 50,
    paddingRight: 50,
  },
  title: {
    alignItems: "center",
    justifyContent: "center",
    fontSize: 18,
    fontWeight: "bold",
  },
  legend: {
    margin: -15
  },
  pie: {
    marginTop: -25
  },
  legendText: {
    marginTop: 3
  },
  legendItem: {
    marginRight: 10,
    justifyContent: "space-between"
  }
  
});

export default PieChart;
