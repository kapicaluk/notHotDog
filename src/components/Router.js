import React from "react";
import { StyleSheet, TouchableHighlight, View } from "react-native";
import { Scene, Router, TabBar, Actions } from "react-native-router-flux";
import { Header, Icon, Title } from 'react-native-elements';
import Signin from '../containers/auth/Signin';
import Signup from '../containers/auth/Signup';
import requireAuth from '../containers/auth/requireAuth';
import requireNotAuth from '../containers/auth/requireNotAuth';
// Containers go here:
import LoggedInLanding from "../scenes/LoggedInLanding"; /* to navigate to camera or account home */
import AccountHome from "../scenes/AccountHome"; /* to display most recent meal */
import AccountSettings from "../scenes/AccountSettings"; /* to edit user account settings */
import History from "../scenes/History"; /* to see all past meal data */
import Camera from "../scenes/Camera"; /* access to the camera */
import NutritionHistory from "../scenes/NutritionHistory";
import FoodSelector from "../scenes/FoodSelector";



const RouterComponent = () => (
  <Router style={styles.container}>
    <Scene 
      key="root"
      navigationBarStyle={styles.navBar}
      nagivationTitleStyle={styles.navBar}
      >
      <Scene key="auth" tabs={true} title="Welcome" hideTabBar>
        <Scene key="signup" component={requireNotAuth(Signup)} />
        <Scene key="signin" component={requireNotAuth(Signin)} initial={true} />
      </Scene>
      <Scene key="settings" component={AccountSettings} title="Account Settings" />
      <Scene key="FoodSelector" component={FoodSelector} title="Select" />
      <Scene key="tabbar" tabs={true} showLabel={false} swipeEnabled={true} tabBarStyle={styles.tabContainer} activeBackgroundColor="#76ffe9"
        renderRightButton={() => {
          return <Icon name="settings" underlayColor="#36d7b7" onPress={() => Actions.settings() } />
        }}>
        <Scene key="AccountHome" title="Your Latest Meal" icon={() => <Icon name="local-dining" />}  component={AccountHome} />
        <Scene key="history" title="Meal Diary" icon={() => <Icon name="date-range" />} component={History} />
        <Scene key="nutrition" title="Nutrition History" icon={() => <Icon name="assessment" />} component={NutritionHistory} />
        <Scene key="camera" title="Camera" component={Camera} icon={() => <Icon name="camera-alt" />} initial={true} />
      </Scene>
    </Scene>
  </Router>
);

const styles = StyleSheet.create({
  container: {
    flex: 1
  },
  navBar: {
    backgroundColor: '#36d7b7',
    padding: 10
  },
  navTitle: {
    color: 'black'
  },
  tabContainer: {
    backgroundColor: '#36d7b7'
  }
})

export default RouterComponent;
