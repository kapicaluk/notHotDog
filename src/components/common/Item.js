/* eslint-disable import/prefer-default-export */
import React from 'react';
import PropTypes from 'prop-types';
import { View } from 'react-native';

const propTypes = {
  children: PropTypes.node.isRequired,
  style: PropTypes.object,
};

const defaultProps = {
  style: {},
};

const Item = props => (
  <View style={[styles.container, props.style]}>
    {props.children}
  </View>
);

const styles = {
  container: {
    alignSelf: 'stretch',
    padding: 5,
    backgroundColor: 'white',
    opacity: 1,
    borderRadius: 5,
    justifyContent: 'flex-start',
    flexDirection: 'row',
    position: 'relative',
    marginRight: 10,
    marginLeft: 10
  }
};

Item.defaultProps = defaultProps;
Item.propTypes = propTypes;

export { Item };
