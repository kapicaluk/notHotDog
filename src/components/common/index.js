export * from './Button';
export * from './Item';
export * from './Container';
export * from './Input';
export * from './Spinner';
export * from './Confirm';
