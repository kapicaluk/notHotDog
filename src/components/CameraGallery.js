import React from 'react';
import { View, StyleSheet } from 'react-native';
import { ImagePicker } from 'expo';
import { dispatch } from 'redux';
import { getOptions } from '../store/food';
import { Spinner } from './common/index';
import clarifaiCall from '../utilities/clarifaiCall';

export default class GalleryScreen extends React.Component {
  render() {

    ImagePicker.launchImageLibraryAsync({base64: true})
    .then(photo => {
      if (photo.cancelled) {
        this.props.onPress();
      } else {
        clarifaiCall(photo.base64, this.props.restrictions, this.props.allergies)
      }
    })

    return (
      <View style={styles.container}>
        <Spinner />
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    paddingTop: 20,
  },
  navBar: {
    backgroundColor: '#36d7b7',
    padding: 10
  },
  navTitle: {
    color: 'black'
  },
});