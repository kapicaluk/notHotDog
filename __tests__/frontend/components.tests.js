import React from 'react';
// import Signin from '../../src/components/auth/Signin';
import {Button, Confirm, Container, Header, Input, Item, Spinner} from '../../src/components/common';
import IconContainer from '../../src/components/IconContainer';

import renderer from 'react-test-renderer';
import { createStore } from 'redux';

/* Auth Components */
// Try these later??

/* Common Components */
describe('Button component', () => {
  it('renders without crashing', () => {
    const renderedButton = renderer.create(<Button onPress={() => {}} children={[]}/>).toJSON();
    expect(renderedButton).toMatchSnapshot();
  })
})

describe('Confirm component', () => {
  it('renders without crashing', () => {
    const renderedConfirm = renderer.create(
      <Confirm 
        onPress={() => {}} 
        children={[]} 
        visible={true} 
        onAccept={() => {}}
        onDecline={() => {}}
      />).toJSON();
    expect(renderedConfirm).toMatchSnapshot();
  })
})

describe('Container component', () => {
  it('renders without crashing', () => {
    const renderedContainer = renderer.create(<Container children={[]}/>).toJSON();
    expect(renderedContainer).toMatchSnapshot();
  })
})

describe('Header component', () => {
  it('renders without crashing', () => {
    const renderedHeader = renderer.create(<Header headerText={''}/>).toJSON();
    expect(renderedHeader).toMatchSnapshot();
  })
})

describe('Input component', () => {
  it('renders without crashing', () => {
    const renderedInput = renderer.create(
      <Input 
        input={{}} 
        meta={{}} 
        placeholder={''}
      />).toJSON();
    expect(renderedInput).toMatchSnapshot();
  })
})

describe('Item component', () => {
  it('renders without crashing', () => {
    const renderedItem = renderer.create(<Item children={[]}/>).toJSON();
    expect(renderedItem).toMatchSnapshot();
  })
})

describe('Spinner component', () => {
  it('renders without crashing', () => {
    const renderedSpinner = renderer.create(<Spinner />).toJSON();
    expect(renderedSpinner).toMatchSnapshot();
  })
})

/* Icon Container */
describe('Icon Container component', () => {
  it('renders without crashing', () => {
    const renderedIconContainer = renderer.create(<IconContainer />).toJSON();
    expect(renderedIconContainer).toMatchSnapshot();
  })
})