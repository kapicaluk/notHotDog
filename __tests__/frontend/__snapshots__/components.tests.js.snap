// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component renders without crashing 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignSelf": "stretch",
      "backgroundColor": "#039be5",
      "borderRadius": 3,
      "flex": 1,
      "marginTop": 10,
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "alignSelf": "center",
          "color": "#fff",
          "fontSize": 16,
          "fontWeight": "600",
          "paddingBottom": 10,
          "paddingTop": 10,
        },
        Object {},
      ]
    }
  />
</View>
`;

exports[`Confirm component renders without crashing 1`] = `
<Modal
  animationType="fade"
  hardwareAccelerated={false}
  onRequestClose={[Function]}
  transparent={true}
  visible={true}
>
  <View
    style={
      Object {
        "backgroundColor": "rgba(0, 0, 0, 0.75)",
        "flex": 1,
        "justifyContent": "center",
        "position": "relative",
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#fff",
            "flexDirection": "row",
            "justifyContent": "flex-start",
            "padding": 5,
            "position": "relative",
          },
          Object {
            "justifyContent": "center",
          },
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "flex": 1,
            "fontSize": 18,
            "lineHeight": 40,
            "textAlign": "center",
          }
        }
      />
    </View>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#fff",
            "flexDirection": "row",
            "justifyContent": "flex-start",
            "padding": 5,
            "position": "relative",
          },
          Object {},
        ]
      }
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignSelf": "stretch",
            "backgroundColor": "#e62117",
            "borderRadius": 3,
            "flex": 1,
            "marginTop": 10,
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "alignSelf": "center",
                "color": "#fff",
                "fontSize": 16,
                "fontWeight": "600",
                "paddingBottom": 10,
                "paddingTop": 10,
              },
              Object {},
            ]
          }
        >
          Yes
        </Text>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#fff",
            "flexDirection": "row",
            "justifyContent": "flex-start",
            "padding": 5,
            "position": "relative",
          },
          Object {},
        ]
      }
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignSelf": "stretch",
            "backgroundColor": "#039be5",
            "borderRadius": 3,
            "flex": 1,
            "marginTop": 10,
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "alignSelf": "center",
                "color": "#fff",
                "fontSize": 16,
                "fontWeight": "600",
                "paddingBottom": 10,
                "paddingTop": 10,
              },
              Object {},
            ]
          }
        >
          No
        </Text>
      </View>
    </View>
  </View>
</Modal>
`;

exports[`Container component renders without crashing 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
        "elevation": 1,
        "marginLeft": 10,
        "marginRight": 10,
        "marginTop": 10,
        "paddingBottom": 10,
        "paddingTop": 10,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.2,
      },
    ]
  }
/>
`;

exports[`Header component renders without crashing 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F8F8F8",
      "elevation": 2,
      "height": 60,
      "justifyContent": "center",
      "paddingTop": 15,
      "position": "relative",
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.2,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 20,
      }
    }
  >
    
  </Text>
</View>
`;

exports[`Icon Container component renders without crashing 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "justifyContent": "flex-end",
    }
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        Array [
          Object {
            "margin": 7,
          },
          Object {
            "borderRadius": 28,
            "height": 52,
            "width": 52,
          },
          Object {
            "shadowColor": "rgba(0,0,0, .4)",
            "shadowOffset": Object {
              "height": 1,
              "width": 1,
            },
            "shadowOpacity": 1,
            "shadowRadius": 1,
          },
          Object {
            "alignItems": "center",
            "backgroundColor": "#00aced",
            "justifyContent": "center",
          },
          undefined,
        ],
      ]
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "white",
            "fontSize": 24,
          },
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            undefined,
          ],
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      îŽ¯
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        Array [
          Object {
            "margin": 7,
          },
          Object {
            "borderRadius": 28,
            "height": 52,
            "width": 52,
          },
          Object {
            "shadowColor": "rgba(0,0,0, .4)",
            "shadowOffset": Object {
              "height": 1,
              "width": 1,
            },
            "shadowOpacity": 1,
            "shadowRadius": 1,
          },
          Object {
            "alignItems": "center",
            "backgroundColor": "#517fa4",
            "justifyContent": "center",
          },
          undefined,
        ],
      ]
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "white",
            "fontSize": 24,
          },
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            undefined,
          ],
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      î¢¸
    </Text>
  </View>
</View>
`;

exports[`Input component renders without crashing 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderBottomWidth": 1,
        "borderColor": "#ddd",
        "flex": 1,
        "flexDirection": "row",
        "height": 40,
      },
      Object {},
    ]
  }
>
  <TextInput
    autoCorrect={false}
    multiline={false}
    onChangeText={[Function]}
    placeholder=""
    secureTextEntry={false}
    style={
      Object {
        "color": "#000",
        "flex": 2,
        "fontSize": 18,
        "lineHeight": 23,
        "paddingLeft": 5,
        "paddingRight": 5,
      }
    }
    value={undefined}
  />
</View>
`;

exports[`Item component renders without crashing 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "padding": 5,
        "position": "relative",
      },
      Object {},
    ]
  }
/>
`;

exports[`Spinner component renders without crashing 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <ActivityIndicator
    animating={true}
    color="#999999"
    hidesWhenStopped={true}
    size="large"
  />
</View>
`;
